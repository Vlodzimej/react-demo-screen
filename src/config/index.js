export const STREAM_URL = '';
export const SOURCE_URI = "rtsp://10.1.0.7:554/ISAPI/streaming/channels/101";
export const WEATHER_URL = 'http://api.openweathermap.org/data/2.5/weather';
export const WEATHER_QUERY_PARAMS = {
    APPID: 'f4e4652a709549a302164efb45ded543',
    id: 553915, // Калуга
    units: 'metric',
    lang: 'ru',
};

export const ID_SERVER_URL = 'http://192.168.3.5:8080';
export const EVENT_SERVICE_ENDPOINT_URL = '/BioxidServer/jsonrpc/eventmonitor';
export const MEDIA_CHANNEL_ID = '3EhLgxzWBbsAEVKdntgd4J';
export const EVENT_CODE = 70208;
export const EMPLOYEE_CARD_LIFETIME = 30000;

export const NVR_SERVER = 'nvr1.algont';
export const VA_SERVER = 'va.algont';

export const WEBSOCKET_VA = 'websocket-va.nvr1.algont';
export const WEBSOCKET_RTSP = 'websocket-rtsp';

export const VA_PORT = '8899';
export const RTSP_PORT = '8898';

export const STREAM_ID = 'welcomeStream';

export const VA_SERVICE_ENDPOINT = 'http://' + VA_SERVER + ':' + VA_PORT + '/service';
export const VA_SYSTEM_ENDPOINT = 'http://' + VA_SERVER + ':' + VA_PORT + '/system';

export const RTSP_SERVICE_ENDPOINT = 'http://' + VA_SERVER + ':' + RTSP_PORT + '/service';
export const RTSP_SYSTEM_ENDPOINT = 'http://' + VA_SERVER + ':' + RTSP_PORT + '/system';

export const WEBSOCKET_VA_SERVICE_ENDPOINT = 'http://' + WEBSOCKET_VA + '/services/va_processor/service';

export const EVENT_SERVICE_ENDPOINT = ID_SERVER_URL + '/BioxidServer/jsonrpc/eventmonitor';

export const BANNER_INTERVAL = 10000;
export const BANNER_AMOUNT = 5;