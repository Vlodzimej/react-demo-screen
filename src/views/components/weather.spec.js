// weather.spec.js
import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Weather from './weather';
import * as config from '../../config';

configure({ adapter: new Adapter() });

describe('Render Weather', () => {

    it('render correctly', () => {
        const wrapper = renderer.create(<Weather {...weatherData}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('render without props', () => {
        const wrapper = renderer.create(<Weather />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('check elements', () => {
        const wrapper = shallow(<Weather {...weatherData}/>);

        const icon = wrapper.find("[name='icon']");
        const temp = wrapper.find("[name='temp']");
        const description = wrapper.find("[name='description']");

        expect(icon).toHaveLength(1);
        expect(temp).toHaveLength(1);
        expect(description).toHaveLength(1);

        const tempText = `+${weatherData.main.temp}°`;
        expect(temp.text()).toEqual(tempText);

        const descriptionText = weatherData.weather[0].description;
        expect(description.text()).toEqual(descriptionText);
    })

    it('check elements without props', () => {
        const wrapper = shallow(<Weather />);
        expect(wrapper.find("img[name='loading']")).toHaveLength(1);
    });
});

const weatherData = {
    coord: { lon: 36.28, lat: 54.53 },
    weather: [
        { id: 500, main: 'Rain', description: 'легкий дождь', icon: '10d' },
    ],
    base: 'stations',
    main: {
        temp: 16,
        pressure: 1018,
        humidity: 93,
        temp_min: 16,
        temp_max: 16,
    },
    visibility: 10000,
    wind: { speed: 2, deg: 330 },
    clouds: { all: 90 },
    dt: 1557918000,
    sys: {
        type: 1,
        id: 9017,
        message: 0.0054,
        country: 'RU',
        sunrise: 1557883822,
        sunset: 1557941534,
    },
    id: 553915,
    name: 'Kaluga',
    cod: 200,
};
