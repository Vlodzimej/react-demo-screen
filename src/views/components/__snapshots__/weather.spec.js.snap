// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Weather render correctly 1`] = `
<div>
  <div
    className="ant-row"
    style={
      Object {
        "marginTop": "32px",
      }
    }
  >
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <div
        name="icon"
      >
        <img
          src="images/weather/10d.png"
        />
      </div>
    </div>
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "display": "flex",
          "fontSize": "6em",
          "justifyContent": "left",
        }
      }
    >
      <div
        name="temp"
      >
        +
        16
        °
      </div>
    </div>
    <div
      className="ant-col ant-col-24"
      style={
        Object {
          "display": "flex",
          "fontSize": "3em",
          "justifyContent": "center",
          "textTransform": "uppercase",
        }
      }
    >
      <div
        name="description"
      >
        легкий дождь
      </div>
    </div>
  </div>
</div>
`;

exports[`Render Weather render without props 1`] = `
<div>
  <img
    name="loading"
    src="images/loading_planet.gif"
    style={
      Object {
        "height": "150px",
        "width": "150px",
      }
    }
  />
</div>
`;
